i3lock.c

```diff
@@ -1009,13 +1009,29 @@ int main(int argc, char *argv[]) {
                                  (uint32_t[]){XCB_EVENT_MASK_STRUCTURE_NOTIFY});
 
     if (image_path) {
-        /* Create a pixmap to render on, fill it with the background color */
-        img = cairo_image_surface_create_from_png(image_path);
-        /* In case loading failed, we just pretend no -i was specified. */
-        if (cairo_surface_status(img) != CAIRO_STATUS_SUCCESS) {
-            fprintf(stderr, "Could not load image \"%s\": %s\n",
-                    image_path, cairo_status_to_string(cairo_surface_status(img)));
-            img = NULL;
+        /* Check file exists and has correct PNG header */
+        if (access(image_path, F_OK) == -1) {
+            fprintf(stderr, "Image file path \"%s\" doesn't exist\n", image_path);
+        } else {
+            FILE* png_file = fopen(image_path, "r");
+            unsigned char png_header[8];
+            memset(png_header, 0, 8);
+            fread(png_header, 1, 8, png_file);
+            fclose(png_file);
+
+            if (png_header[0] != 137 || png_header[1] != 80 || png_header[2] != 78 || png_header[3] != 71
+                        || png_header[4] != 13 || png_header[5] != 10 || png_header[6] != 26 || png_header[7] != 10) {
+                fprintf(stderr, "Provided PNG image \"%s\" is incorrect\n", image_path);
```

Comment:
Let’s make this error message more descriptive. How about:

File "%s" does not start with a PNG header. i3lock currently only supports loading PNG files.